### DISCUSSION TITLE
Running a batch script when power adapter is disconnected.

### DISCUSSION THREAD

**Discussion by MPA10** [BartoszCichecki/LenovoLegionToolkit#995] [Q&A] [answered]
Hi everyone,

I've created a batch script to terminate certain apps and processes, including their child processes, to help conserve battery power. The script runs successfully when I execute it from File Explorer. However, when I add a 'Run' step to my actions and specify the full file path in the 'Executable Path' field, nothing happens.

Do I need to provide additional arguments? What is the correct method to make this work? I couldn't find any relevant information in the documentation so far.

Thanks in advance.

**Comment by MPA10** [comment]
I wanted to provide an update regarding my previous message. It turns out the batch script I created does work as intended. The reason I initially thought it wasn't functioning was that the script executed without opening the Command Prompt window. 

Why is the Command Prompt window not opening when the script is executed from another program, compared to when it's run from File Explorer?

**Reply by BartoszCichecki** [reply, marked as answer]
The reason is that LLT explicitly runs scripts in "no window" mode :)
